lib/frameworks/jasmine2.js

```diff
@@ -74,11 +74,29 @@ exports.run = function(runner, specs) {
   var reporter = new RunnerReporter(runner);
   jasmine.getEnv().addReporter(reporter);
 
-  // Filter specs to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
-  jasmine.getEnv().specFilter = function(spec) {
+  // Filter specs/suite to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
+  jasmine.getEnv().specFilter = function(spec, suite) {
+    var suiteMatched = isSuiteMatched(suite);
+
+    //If it matched at suite level and invert is true, all the specs will not be included.
+    if(suiteMatched && jasmineNodeOpts.invertGrep){
+      spec.pend();
+      return true;
+    }
+
+    //If it matched at suite level and invert is false, all the specs will be included.
+    if(suiteMatched && !jasmineNodeOpts.invertGrep){
+      return true;
+    }
+
+    var specWithSuiteFullName = spec.getFullName();
+    if(suite){
+      specWithSuiteFullName += suite.getFullName();
+    }
+
     var grepMatch = !jasmineNodeOpts ||
         !jasmineNodeOpts.grep ||
-        spec.getFullName().match(new RegExp(jasmineNodeOpts.grep)) != null;
+        specWithSuiteFullName.match(new RegExp(jasmineNodeOpts.grep, 'i')) != null;
```

Comment:
This (both adding the suite name and changing to case-insensitivity) is a breaking change.